tests/Propulsion.MessageDb.Integration/Tests.fs

```diff
@@ -78,3 +88,47 @@ let ``It processes events for a category`` () = task {
     // they were handled in order within streams
     let ordering = handled |> Seq.groupBy fst |> Seq.map (snd >> Seq.map snd >> Seq.toArray) |> Seq.toArray
     test <@ ordering |> Array.forall ((=) [| 0L..19L |]) @> }
+
+type LogSink() =
+    let mutable items = 0
+    let mutable calledTimes = 0
+    member _.Events = items
+    member _.CallCount = calledTimes
+    interface ILogEventSink with
+        member _.Emit(logEvent) =
+            match logEvent with
+            | Propulsion.Feed.Core.Log.MetricEvent (Propulsion.Feed.Core.Log.Metric.Read r) ->
+                calledTimes <- calledTimes + 1
+                items <- items + r.items
+            | _ -> ()
+
+[<Fact>]
+let ``It doesn't read the tail event again`` () = task {
+    let logSink = LogSink()
+    let log = LoggerConfiguration().WriteTo.Sink(logSink).CreateLogger()
+    let consumerGroup = $"{Guid.NewGuid():N}"
+    let category = $"{Guid.NewGuid():N}"
+    let connString = "Host=localhost; Database=message_store; Port=5433; Username=message_store; Password=;"
+    use conn = new NpgsqlConnection(connString)
```

Comment:
extract 112-113, 115 (connstr, use conn, OpenAsync) as a 'PrepareStore' that takes a NpgslConn -> Task, to share with other test ?